packages/ai/core/generate-image/generate-image.ts

```diff
@@ -110,20 +110,27 @@ Only applicable for HTTP-based providers.
 }): Promise<GenerateImageResult> {
   const { retry } = prepareRetries({ maxRetries: maxRetriesArg });
 
+  // extract maxImagesPerCall from providerOptions as it's not meant to be passed to `doGenerate()`
+  const [
+    [providerName, { maxImagesPerCall, ...generateProviderOptions } = {}] = [],
+  ] = Object.entries(providerOptions ?? {});
```

Comment:
I'm surprised that we don't know `providerName` at the time when `generateImage()` is called. Can't we somehow reliable extract it from `model`?

I also don't know if this change breaks a use case that I'm not aware of?